v2.2/cluster-setup-troubleshooting.md

```diff
@@ -6,202 +6,434 @@ toc: true
+-   It also explains to future administrations why your node IDs have gaps (e.g., why the nodes are numbered n1, n2, and n8).
+
+You can follow the discussion here: [https://github.com/cockroachdb/cockroach/issues/24636](https://github.com/cockroachdb/cockroach/issues/24636)
+
+## Replication issues
+
+### Admin UI shows under-replicated/unavailable ranges
+
+An under-replicated range is one in which the number of “up to date” replicas is below the desired replication. An unavailable range is one in which the number of “up to date” replicas is below quorum. When a Cockroach node dies (or is partitioned) the under-replicated range count will briefly spike while the system recovers.
+
+**Explanation:** Cockroach uses consensus replication and requires a quorum of the replicas to be available in order to allow both writes and reads to the range. The number of failures that can be tolerated is equal to (Replication factor - 1)/2. Thus CockroachDB requires (n-1)/2 nodes to achieve quorum. is For example, with 3x replication, one failure can be tolerated; with 5x replication, two failures, and so on.
+
+-   Under-replicated Ranges: When a cluster is first initialized, the few default starting ranges will only have a single replica, but as soon as other nodes are available, they will replicate to them until they've reached their desired replication factor. If a range does not have enough replicas, the range is said to be "under-replicated".
 
-**Explanation**: When starting a node, the directory you choose to store the data in also contains metadata identifying the cluster the data came from. This causes conflicts when you've already started a node on the server, have quit `cockroach`, and then tried to join another cluster. Because the existing directory's cluster ID doesn't match the new cluster ID, the node cannot join it.
+-   Unavailable Ranges: If a majority of a range's replicas are on nodes that are unavailable, then the entire range is unavailable and will be unable to process queries.
 
-### Replication error in a multi-node cluster
+**Solution:**
 
-If data is not being replicated to some nodes in the cluster, we recommend checking out the following:
+To identify under-replicated/unavailable ranges:
 
-- Ensure every node but the first was started with the `--join` flag set to the hostname and port of first node (or any other node that's successfully joined the cluster).
+1.  Access the Admin UI.
 
-    If the flag was not set correctly for a node, shut down the node and restart it with the `--join` flag set correctly. See [Stop a Node](stop-a-node.html) and [Start a Node](start-a-node.html) for more details.
+2.  On the **Cluster Overview** page, check the **Replication Status**. If the **Under-replicated ranges** or **Unavailable ranges** count is non-zero, then you have under-replicated or unavailable ranges in your cluster.
```

Comment:
Step 3 here would probably be to check the network latency report (#/reports/network) and ensure there's no network partition. If there's no partition, and there's still no upreplication after 5 mins, then they should file an issue.



--

Reply:
Done.